### ISSUE TITLE
apertium-tagger fails for some pairs on some linux distros

### ISSUE THREAD

**Issue by jonorthwash** [apertium/apertium#63]
The modes `eng-cat` and `eng-deu` appear to crash `apertium-tagger` on CentOS 7, and fail on Fedora 30.

See [original report of this issue](https://github.com/apertium/apertium-all/issues/5).

**Comment by MarcRiera** [comment]
`apertium-eng` uses the perceptron tagger, perhaps it has something to do with it.

**Comment by jonorthwash** [comment]
Yes, I would say that's likely.  But the following behaviour reported by @TinoDidriksen suggests that whether or not it's related to that, there's a major problem that needs to be fixed:
> I can reproduce the bad behavior ... on CentOS 7, and on Fedora 30 it gives no output whatsoever. Works flawlessly on Debian/Ubuntu.

**Comment by PRISHIta123** [comment]
Hi, I was just going through the apertium-tagger model and I was wondering if it could have something to do with the error gradient exploding exponentially or diminishing (also known as the vanishing gradient problem) as it is propagated backward through the multilayer perceptron network while training. This may lead to the gradient computation occupying too much memory at training time, that could be the reason behind the crash.

**Comment by unhammer** [comment]
@PRISHIta123 they're crashing at run-time, not during training
